src/updatecheck/UpdateChecker.cpp

```diff
@@ -84,47 +89,59 @@ void UpdateChecker::fetchFinished()
 
         if (!jsonObject.value("tag_name").isUndefined()) {
             version = jsonObject.value("tag_name").toString();
-            hasNewVersion = compareVersions(version, QString(KEEPASSXC_VERSION));
+            hasNewVersion = compareVersions(QString(KEEPASSXC_VERSION), version);
         }
+
+        // Check again in 7 days
+        // TODO: change to toSecsSinceEpoch() when min Qt >= 5.8
+        config()->set("GUI/CheckForUpdatesNextCheck", Clock::currentDateTime().addDays(7).toTime_t());
     } else {
         version = "error";
     }
 
     emit updateCheckFinished(hasNewVersion, version, m_isManuallyRequested);
 }
 
-bool UpdateChecker::compareVersions(const QString& remoteVersion, const QString& localVersion)
+bool UpdateChecker::compareVersions(const QString& localVersion, const QString& remoteVersion)
 {
+    // Quick full-string equivalence check
     if (localVersion == remoteVersion) {
-        return false; // Currently using updated version
+        return false;
     }
 
-    QRegularExpression verRegex("^(\\d+(\\.\\d+){0,2})(-\\w+)?$", QRegularExpression::CaseInsensitiveOption);
+    QRegularExpression verRegex(R"(^((?:\d+\.?){3})(-\w+?)?(\d+)?$)");
```

Comment:
This regex looks a bit weird. Why not

```
^((?:\d+\.){2}\d+)-(\w+?)(\d+)?$
```
This matches the version number more accurately and by not capturing the hyphen you can use exact matching instead of `contains()` for betas and snapshots. You could even capture each of the version number components individually instead of splitting them afterwards.

Reply:
if you do not make the hyphen optional then a plain version number (2.4.0) will not match. I was considering making a group for each digit, but we are really splitting hairs here.

Reply:
The hyphen can be made optional by putting it all in a non-capturing group:
```
^((?:\d+\.){2}\d+)(?:-(\w+?)(\d+))?$
```

Reply:
Changes made